File: backend/app/models/task.py
"""
Task model for todo items
Task: T-007 - Create Task SQLModel
"""

from sqlmodel import SQLModel, Field
from datetime import datetime
from typing import Optional


class Task(SQLModel, table=True):
    """
    Task model representing todo items.
    Each task belongs to a specific user (identified by Better Auth UUID).
    """

    __tablename__ = "tasks"

    id: Optional[int] = Field(default=None, primary_key=True)
    user_id: str = Field(index=True, max_length=255)  # Better Auth UUID
    title: str = Field(max_length=200)
    description: str = Field(default="")
    completed: bool = Field(default=False, index=True)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

File: frontend/types/task.ts
/**
 * Task type definitions
 * Task: T-020 - Create Task Type Definitions
 */

export interface Task {
  id: number
  user_id: string
  title: string
  description: string
  completed: boolean
  created_at: string
  updated_at: string
}

export interface CreateTaskRequest {
  title: string
  description?: string
}

export interface UpdateTaskRequest {
  title?: string
  description?: string
}
